### DISCUSSION TITLE
Build issue with VS2019

### DISCUSSION THREAD

**Discussion by Emma1963** [winfsp/winfsp#582] [Q&A] [answered]
Hi,
I want to build winfsp version 1.12 with VS2022+SDK 261000. It could not be built for x86.
I searched about the issue. It seems 2022 SDK doesn't support x86.
I would like to ask what version of SDK and WDK I should select to build winfsp with VS2019.

I would appreciate it if you could share with me your environment for building.

Thank you.

**Comment by Emma1963** [comment, marked as answer]
I got my answer.
1. VS2022 with the latest SDK/WDK 26100 does not work. It won't support build as x86. It also changed the support version. Some definitions can't be found in the SDK. I tried to modify "NTDDI_VERSION=0x06010000;_WIN32_WINNT=0x0601" to match the SDK, and then it can build but still have other issues.
2. I use VS 2019 and SDK/WDK 22000. In the beginning, it didn't work. I got a pop exception every time I opened VS2019
 "Driver Testing Could not load test information.  “TaefTestParser, Version=2.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35”

The reason is I installed two versions of WDK, one for VS2022, and one for VS2019. So I uninstall all two versions of WDK and related SDK.
Install SDK and WDK for VS2019. Then I can build it successfully with x64, x86 and x64 ARM

**Reply by billziss-gh** [reply]
Unfortunately only VS2019 works to build WinFsp. The primary issue is that Microsoft has removed support for x86 and Windows versions before Windows 10 in the latest WDK.

See here for more: https://github.com/winfsp/winfsp/issues/435

**Comment by Emma1963** [comment]
@billziss-gh Thank you.
One more question. We rebranded one version of v 1.4 before and used winfsp in our app.
Now we want to explore supporting ARM. So we want to sync Winfsp to version 1.12 first. I noticed winsfp has made the rebrand easier now.
Just modify the information in build.version.props.

My question here is about the MyFspFsctlDeviceClassGuid. I understood we need to create a new one for our rebrand let us say CFsp. But as we already used CFsp (winfsp version1.4). Is it ok we just modify these two Guids by creating new ones? Will it affect the old CFso 1.4 we used before? Will it affect our app installer to uninstall CFsp and install new version CFsp(version 1.12)? 
```
<!-- When rebranding WinFsp you MUST change the following GUIDs - use VS "Create GUID" tool -->
        <MyFspFsctlDeviceClassGuid>{ 0x6f9d25fa, 0x6dee, 0x4a9d, { 0x80, 0xf5, 0xe9, 0x8e, 0x14, 0xf3, 0x5e, 0x54 } }</MyFspFsctlDeviceClassGuid>
        <MyFspFsvrtDeviceClassGuid>{ 0xb48171c3, 0xdd50, 0x4852, { 0x83, 0xa3, 0x34, 0x4c, 0x50, 0xd9, 0x3b, 0x17 } }</MyFspFsvrtDeviceClassGuid>
```

**Reply by billziss-gh** [reply]
These GUIDs are used to assign default security permissions for the WinFsp drivers in the registry. If you rebrand WinFsp you must change them, otherwise there could be clashes with official WinFsp releases (or even with other peoples’ rebranded WinFsp releases who have failed to change these GUID’s).

I do not know how this affects your existing rebranding scheme. However the actual GUID’s are not very important and can change across releases.

Please make sure to read and understand the full details in https://github.com/winfsp/winfsp/wiki/WinFsp-Rebranding. In particular make sure that GUID’s in your installer do not clash with WinFsp ones (e.g. by using a tool such as https://github.com/winfsp/winfsp/blob/master/tools/gensrc/wixguid.py).

**Reply by Emma1963** [reply]
Thank you so much.

**Comment by Emma1963** [comment]
@billziss-gh I merged the code to version 1.12, rebranded the name(We have a commercial license), and built success.
I saw the signing process using EV certificate and Windows Partner. (https://github.com/winfsp/winfsp/blob/release/1.12/doc/WinFsp-Building.asciidoc)
1. However, I would like to know if there is a way I can test with a test sign/without a sign before I do the sign process. I could not use bcdedit command because of company policy.
2. I tried to use memfs-dotnet to run the test. I could debug memfs-donet and "<rebrandname>fsp-msil.dll". But I could not debug "<rebrandname>fsp-x64.dll". When I run memfs-donet with parameters. I got the error "-2147024319" which I could not figure out the reason. I didn't sign it by EV or disable Windows sign. Is that the reason? 

In general, I would like to know 2 things:
1. Any other way to test.sys file and <rebrandname>fsp-x64.dll without sign process for debug test purposes.
2. How to debug <rebrandname>fsp-x64.dll?
